src/tests/ftest/container/boundary.py

```diff
@@ -22,120 +22,142 @@ class BoundaryTest(TestWithServers):
+    def __init__(self, *args, **kwargs):
+        """Initialize a BoundaryTest object."""
+        super().__init__(*args, **kwargs)
+        self.with_io = False
+        self.io_run_time = None
+        self.io_rank = None
+        self.io_obj_classs = None
+
+    def setUp(self):
+        """Set Up BoundaryTest"""
+        super().setUp()
+        self.pool = []
+        self.with_io = self.params.get("with_io", '/run/boundary_test/*')
+        self.io_run_time = self.params.get("run_time", '/run/container/execute_io/*')
+        self.io_rank = self.params.get("rank", '/run/container/execute_io/*')
+        self.io_obj_classs = self.params.get("obj_classs", '/run/container/execute_io/*')
+
+    def get_pool(self, *args, **kwargs):
+        """Get a test pool object and append to list.
+
+        Returns:
+            TestPool: the created test pool object.
+
+        """
+        pool = super().get_pool()
+        self.pool.append(pool)
+        return pool
+
+    def create_container_and_test(self, pool=None, cont_num=1):
         """To create single container on pool.
+
         Args:
             pool (str): pool handle to create container.
-            pool_num (int): pool number.
             container_num (int): container number to create.
-            with_io (bool): enable container test with execute_io.
-        """
 
+        """
         try:
             container = self.get_container(pool)
         except (DaosTestError, TestFail) as err:
-            msg = "#(3.{}.{}) container create failed. err={}".format(pool_num, cont_num, err)
+            msg = "#(3.{}.{}) container create failed. err={}".format(pool.label, cont_num, err)
             self.fail(msg)
 
-        self.log.info("===(3.%d.%d)create_container_and_test, container %s created..",
-            pool_num, cont_num, container)
-        if with_io:
-            io_run_time = self.params.get("run_time", '/run/container/execute_io/*')
-            rank = self.params.get("rank", '/run/container/execute_io/*')
-            obj_classs = self.params.get("obj_classs", '/run/container/execute_io/*')
+        self.log.info("===(3.%s.%d)create_container_and_test, container %s created..",
+            pool.label, cont_num, container)
+        if self.with_io:
             try:
-                data_bytes = container.execute_io(io_run_time, rank, obj_classs)
+                data_bytes = container.execute_io(
+                    self.io_run_time, self.io_rank, self.io_obj_classs)
                 self.log.info(
-                    "===(3.%d.%d)Wrote %d bytes to container %s", pool_num, cont_num,
+                    "===(3.%s.%d)Wrote %d bytes to container %s", pool.label, cont_num,
                     data_bytes, container)
             except (DaosTestError, TestFail) as err:
-                msg = "#(3.{}.{}) container IO failed, err: {}".format(pool_num, cont_num, err)
+                msg = "#(3.{}.{}) container IO failed, err: {}".format(pool.label, cont_num, err)
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string